src/mmw/apps/modeling/mapshed/tasks.py

```diff
@@ -312,6 +318,10 @@ def nlcd_soils(sjs_result):
         g2 = settings.SOIL_GROUP[g]
         ng_count[(n, g2)] = count + ng_count.get((n, g2), 0)
 
+    # Immediately return an error for AOIs without enough land cover
+    if len(nt_count.values()) == 0:
```

Comment:
Use `n_count` instead of `nt_count`, since we are looking for NLCD values only, not how they pair with soil texture.

Can we test `ngt_count` itself? It might be quicker to test that instead of until after destructuring it.

Also, try phrasing the comment without "AOI". This function simply processes a dictionary, and is not aware of concepts like area of interest. Perhaps something like "Raise exception if no NLCD values".